docs/howtos/integrations/_langgraph_agent_evaluation.md

```diff
@@ -0,0 +1,550 @@
+# Building and Evaluating a ReAct Agent for Fetching Metal Prices
+
+AI agents are becoming increasingly valuable in domains like finance, e-commerce, and customer support. These agents can autonomously interact with APIs, retrieve real-time data, and perform tasks that align with user goals. Evaluating these agents is crucial to ensure they are effective, accurate, and responsive to different inputs.
+
+In this tutorial, we'll:
+
+1. Build a [ReAct agent](https://arxiv.org/abs/2210.03629) to fetch metal prices.
+2. Set up an evaluation pipeline to track key performance metrics.
+3. Run and assess the agent's effectiveness with different queries.
+
+Click the [link](https://colab.research.google.com/drive/1IsmUfCVD_pVQ7HGsKj97BXOZBGXTIF8m?usp=sharing) to open the notebook in Google Colab.
+
+## Prerequisites
+- Python 3.8+
+- OpenAI API key
+- Basic understanding of LangGraph, LangChain and LLMs
+
+## Installing Ragas and Other Dependencies
+Install Ragas and Langgraph with pip:
+
+
+```python
+%pip install langgraph==0.2.44
+%pip install ragas
+%pip install nltk
+```
+
+## Building the ReAct Agent
+
+### Initializing the External Components: Setting Up the Metals.Dev API
+In this section, we'll walk through how to set up the [metals.dev](https://metals.dev/) API, which provides real-time prices for both precious and industrial metals. The API gives us a simple way to fetch the latest prices for metals like gold, silver, copper, and more.
+
+We selected metals.dev due to its free tier, which provides up to 250 requests per month — a sufficient allowance for this educational tutorials. Below are the steps to get started with the service.
+
+Before you can use the [metals.dev](https://metals.dev/) API, you will need to create an account and get an API key. Here’s how to do that:
+
+1. Go to the metals.dev website.
+2. Sign up for a free account by clicking on the Sign Up button.
+3. Once you've signed up and logged in, you will be taken to your dashboard where you can find your API key.
```

Comment:
I'm not sure if this is a good idea - this adds another friction point for users - now they have to sign up to another service. You had said you have a mock file right. Lets use that and just have a note pointing to the site?

Reply:
the goal is to make sure the users come to evaluations as fast as possible.

Reply:
I understand your concern, and that’s exactly why I made the account setup optional. The idea was to give users the choice: they can try the evaluation without creating an account, but for those who do, it would make their efforts much easier and more beginner-friendly.

Would you like me to remove the section about connecting to the API?

Reply:
yep do remove it - we want to help users spend more time on eval side of things